src/store/hooks/useAuth.tsx

```diff
@@ -0,0 +1,212 @@
+import {
+  createContext, useContext, useMemo, ReactNode,
+  useEffect, useState,
+} from 'react';
+import {
+  getAuth, onAuthStateChanged, User, signOut, Auth,
+} from 'firebase/auth';
+import { LoadingOverlay } from '@mantine/core';
+import { useStorageEngine } from '../storageEngineHooks';
+import { FirebaseStorageEngine } from '../../storage/engines/FirebaseStorageEngine';
+import { GlobalConfig } from '../../parser/types';
+
+interface LocalStorageUser {
+  name: string,
+  email: string
+}
+
+type UserOptions = User | LocalStorageUser | null;
+
+interface UserWrapped {
+  user: UserOptions,
+  determiningStatus: boolean,
+  isAdmin: boolean,
+  adminVerification:boolean
+}
+
+// Defines default AuthContextValue
+interface AuthContextValue {
+  user: UserWrapped;
+  logout: () => void;
```

Comment:
`logout` is `async` below. Do we need to modify the type here?

Reply:
The compiler isn't having a problem with it, but not sure why tbh. I changed to the following:

In  interface AuthContextValue: 
From `logout: () => void` to `logout: () => Promise<void>`

In Auth Context initialization:
From `logout: () => {};` to `logout: async () => {};`